java/memory/memory-core/src/main/java/org/apache/arrow/memory/Accountant.java

```diff
@@ -72,13 +73,13 @@ public Accountant(Accountant parent, String name, long reservation, long maxAllo
     this.reservation = reservation;
     this.allocationLimit.set(maxAllocation);
 
-    if (reservation != 0) {
+    if (reservation != 0 && parent != null) {
```

Comment:
`parentAllocator` looks like to be nullable at https://github.com/davisusanibar/arrow/blob/2092e18752a9c0494799493b12eb1830052217a2/java/memory/memory-core/src/main/java/org/apache/arrow/memory/RootAllocator.java#L61

Reply:
[CheckerFramework](https://checkerframework.org/manual/#introduction:~:text=A%20%E2%80%9Cchecker%E2%80%9D%20is%20a%20compile%2Dtime%20tool) is a compile-time tool, for that reason that condition `&& parent != null` is needed to be added.

Obviously this in checked also at runtime validation for `Preconditions.checkArgument` and also throw `throw new IllegalStateException` if needed.

Reply:
For later investigation, it might be good to see if we can have Preconditions annotated in a way that Checker Framework can understand them.

Reply:
It's supported as of 3 weeks ago https://github.com/typetools/checker-framework/pull/6311

@davisusanibar can you file a follow up issue?

Reply:
That said the condition here is a little too complex, so I don't think it will fix this. (Maybe it's smart and can unpack the assertion above, though!)